gridtools/make_topog/make_topog.py

```diff
@@ -0,0 +1,250 @@
+              help = "")
+# bowl
+@click.option("--bowl_south",
+              type = int,
+              default = 60,
+              help = "")
+@click.option("--bowl_north",
+              type = int,
+              default = 70,
+              help = "")
+@click.option("--bowl_west",
+              type = int,
+              default = 0,
+              help = "")
+@click.option("--bowl_east",
+              type = int,
+              default = 20,
+              help = "")
+# box channel
+@click.option("--jwest_south",
+              type = int,
+              default = 0,
+              help = "")
+@click.option("--jwest_north",
+              type = int,
+              default = 0,
+              help = "")
+@click.option("--ieast_south",
+              type = int,
+              default = 0,
+              help = "")
+@click.option("--ieast_north",
+              type = int,
+              default = 0,
+              help = "")
+# dome
+@click.option("--dome_slope",
+              type = int,
+              default = 0,
+              help = "")
+@click.option("--dome_bottom",
+              type = int,
+              default = 0,
+              help = "")
+@click.option("--dome_embayment_west",
+              type = int,
+              default = 0,
+              help = "")
+@click.option("--dome_embayment_east",
+              type = int,
+              default = 0,
+              help = "")
+@click.option("--dome_embayment_south",
+              type = int,
+              default = 0,
+              help = "")
+@click.option("--dome_embayment_depth",
+              type = int,
+              default = 0,
+              help = "")
+def make_topog(mosaic, topog_type, x_refine, y_refine, bottom_depth, min_depth, scale_factor, topog_file, topog_field,
+               num_filter_pass, flat_bottom, fill_first_row, filter_topog, full_cell, dont_fill_isolated_cells,
+               dont_change_landmask, kmt_min, dont_adjust_topo, fraction_full_cell, dont_open_very_this_cell,
+               min_thickness, rotate_poly, on_grid, round_shallow, fill_shallow, deepen_shallow,
+               smooth_topo_allow_deepening, vgrid_file, gauss_amp, gauss_scale, slope_x, slope_y, bowl_south,
+               bowl_north, bowl_west, bowl_east, jwest_south, jwest_north, ieast_south, ieast_north, dome_slope,
+               dome_bottom, dome_embayment_west, dome_embayment_east, dome_embayment_south, dome_embayment_depth, output):
+
+    # check valid mosaic path and get tiles
+    if(not Path(mosaic).exists()):
+        print("Invalid path given for mosaic file. Exiting...")
+        exit(1)
+    with xarray.open_dataset(mosaic) as ds:
+        if 'ntiles' in ds.dims:
+            ntiles = ds.sizes['ntiles']
+        else:
+            ntiles = 1
+
+    # read in object fields from file
+    mosaicGrid = "placeHolderForMosaicGridConstructor()"
+
+    # create new TopogStruct for output
+    topogOut = TopogObj(output_name=output, ntiles=ntiles)
```

Comment:
could mosaicGrid be made as a class attribute?

Reply:
Yeah that can be added to the class once the mosaic struct is in the code.